File: pubsub/src/main/java/glitch/pubsub/events/SubscribersOnlyEvent.java
package glitch.pubsub.events;

import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import glitch.pubsub.Topic;
import glitch.pubsub.object.json.ModeratorActivation;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class SubscribersOnlyEvent extends AbstractMessageEvent<ModeratorActivation> implements IEvent<GlitchPubSub> {
    public SubscribersOnlyEvent(GlitchPubSub client, Topic topic, ModeratorActivation data) {
        super(client, topic, data);
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/SubscribersOnlyEvent.kt
package glitch.pubsub.events

import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub
import glitch.pubsub.Topic
import glitch.pubsub.`object`.json.ModeratorActivation

class SubscribersOnlyEvent(client: GlitchPubSub, topic: Topic, data: ModeratorActivation) : AbstractMessageEvent<ModeratorActivation>(client, topic, data), IEvent<GlitchPubSub> {

}
